src/inet/TCPEndPointImplLwIP.cpp

```diff
@@ -351,7 +351,7 @@ CHIP_ERROR TCPEndPointImplLwIP::DriveSendingImpl()
             {
                 VerifyOrDie(!startOfUnsent.buffer.IsNull());
 
-                uint16_t bufDataLen = startOfUnsent.buffer->DataLength();
+                uint16_t bufDataLen = static_cast<uint16_t>(startOfUnsent.buffer->DataLength());
```

Comment:
assert for overrun

Reply:
The reason I left out the pre-check in LwIP is because LwIP packet buffers are limited to uint16 size limit.